app/services/db/doc_auth_log/drop_off_rates_helper.rb

```diff
@@ -1,57 +1,89 @@
 module Db
   module DocAuthLog
-    class DropOffRates
+    module DropOffRatesHelper
       STEPS = %w[welcome upload_option front_image back_image ssn verify_info doc_success phone
                  encrypt personal_key verified].freeze
 
-      def call(start, finish)
+      private
+
+      attr_reader :start, :finish, :issuer, :results
+
+      def drop_off_rates(title:, issuer: nil, start: nil, finish: nil)
+        @title = title
+        @issuer = issuer
         @start = start
         @finish = finish
         generate_report
       end
 
-      private
+      def select_count_from_profiles_where_verified_and_active
+        <<~SQL
+          select count(*)
+          from profiles
+          where verified_at is not null and active=true
+        SQL
+      end
+
+      def select_counts_from_doc_auth_logs
+        <<~SQL
+          select count(welcome_view_at) as welcome, count(upload_view_at) as upload_option,
+          count(COALESCE(front_image_view_at,mobile_front_image_view_at)) as front_image,
+          count(COALESCE(back_image_view_at,mobile_back_image_view_at,capture_mobile_back_image_view_at)) as back_image,
+          count(ssn_view_at) as ssn,
+          count(verify_view_at) as verify_info,
+          count(doc_success_view_at) as doc_success,
+          count(verify_phone_view_at) as phone,
+          count(encrypt_view_at) as encrypt,
+          count(verified_view_at) as personal_key
+          from doc_auth_logs
+        SQL
+      end
 
-      attr_reader :start, :finish, :results
+      def at_least_one_image_submitted
+        <<~SQL
+          (front_image_submit_count>0 or back_image_submit_count>0 or mobile_front_image_submit_count>0 or  mobile_back_image_submit_count>0 or  capture_mobile_back_image_submit_count>0)
```

Comment:
the spacing is really inconsistent here and it's a super long line. WDYT of something like:

```ruby
predicates = [
  'front_image_submit_count>0',
  'back_image_submit_count>0',
  'mobile_back_image_submit_count>0',
  'capture_mobile_back_image_submit_count>0',
].join(' or ')

"(#{predicates})"
```

Reply:
nice